File: java/test/com/github/kornilova_l/algorithm_synthesis/tiles/ColouringProblemTest.java
package com.github.kornilova_l.algorithm_synthesis.tiles;

import com.github.kornilova_l.algorithm_synthesis.colouring.ColouringProblem;
import org.apache.commons.io.FileUtils;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.IOException;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;

class ColouringProblemTest {
    private final TileSet tiles32 = new TileGenerator(3, 2, 1).getTileSet();
    private final TileSet tiles23 = new TileGenerator(2, 3, 1).getTileSet();

    @Test
    void toDimacs() throws IOException {
        // TODO: compare set of strings
        /* note: this test may fail because order of lines matters */
        String expected = String.join("\n",
                FileUtils.readLines(new File("java/test_resources/dimacs_4-colouring_2-2-1.txt"), (String) null)) + "\n";
        HashMap<Tile, HashSet<Tile>> graph = new TileGraphBuilder(tiles32, tiles23).getGraph();
        Map<Tile, Integer> ids = ColouringProblem.Companion.assignIds(graph);
        String actual = ColouringProblem.Companion.toDimacs(
                graph, ids, 4
        );
        assertEquals(expected, actual);
    }

}

File: src/test/kotlin/com/github/kornilova_l/algorithm_synthesis/tiles/ColouringProblemTest.kt
package com.github.kornilova_l.algorithm_synthesis.tiles

import com.github.kornilova_l.algorithm_synthesis.grid2D.colouring.ColouringProblem
import com.github.kornilova_l.algorithm_synthesis.grid2D.tiles.TileGenerator
import com.github.kornilova_l.algorithm_synthesis.grid2D.tiles.collections.SimpleTileGraph
import org.apache.commons.io.FileUtils
import org.junit.Assert.assertEquals
import org.junit.Test
import java.io.File
import java.io.IOException
import java.lang.String.join


internal class ColouringProblemTest {
    private val tiles32 = TileGenerator(3, 2, 1).tileSet
    private val tiles23 = TileGenerator(2, 3, 1).tileSet

    @Test
    @Throws(IOException::class)
    fun toDimacs() {
        // TODO: compare set of strings
        /* note: this test may fail because order of lines matters */
        val expected = join("\n",
                FileUtils.readLines(File("src/test/resources/dimacs_4-colouring_2-2-1.txt"), null as String?)) + "\n"
        val graph = SimpleTileGraph(tiles32, tiles23)
        val actual = ColouringProblem.Companion.toDimacs(
                graph, 4
        )
        assertEquals(expected, actual)
    }

}